File: factorial.py
'''
Author:Amala Manoj
Date:29-11-2024
def factorial(n):
    if n==1:
        return 1
    else:
        return n*factorial(n-1)
f=int(input("Enter a number:"))
n=factorial(f)
print("factorial of given number",f,"is",n)

File: factorial.c
// factorial using while loop
#include<stdio.h>
int main()
{
int num1,i,fact=1;
printf("Enter the number:");
scanf("%d",&num1);
i=num1;
if(num1<0)
 {
  	printf("Invalid Entry");
 }
 else
 {
 	while(i>0)
 	{
 	fact=fact*i;
 	i--;
 	}
 	printf("The factorial is %d",fact);
 }
 
 
 
 
 return 0;
}
 
